cumulus/parachains/integration-tests/emulated/tests/coretime/coretime-westend/src/tests/teleport.rs

```diff
@@ -0,0 +1,214 @@
+	test_relay_is_trusted_teleporter,
+};
+
+#[test]
+fn teleport_via_limited_teleport_assets_from_and_to_relay() {
+	let amount = WESTEND_ED * 10;
+	let native_asset: Assets = (Here, amount).into();
+
+	test_relay_is_trusted_teleporter!(
+		Westend,               // Origin
+		vec![CoretimeWestend], // Destinations
+		(native_asset, amount),
+		limited_teleport_assets
+	);
+
+	test_parachain_is_trusted_teleporter_for_relay!(
+		CoretimeWestend, // Origin
+		Westend,         // Destination
+		amount,
+		limited_teleport_assets
+	);
+}
+
+// TODO: Fix transfer_assets weights
+// #[test]
+// fn teleport_via_transfer_assets_from_and_to_relay() {
+// 	let amount = WESTEND_ED * 10;
+// 	let native_asset: Assets = (Here, amount).into();
+
+// 	test_relay_is_trusted_teleporter!(
+// 		Westend,                  // Origin
+// 		vec![CoretimeWestend], // Destinations
+// 		(native_asset, amount),
+// 		transfer_assets
+// 	);
+
+// 	test_parachain_is_trusted_teleporter_for_relay!(
+// 		CoretimeWestend, // Origin
+// 		Westend,            // Destination
+// 		amount,
+// 		transfer_assets
+// 	);
+// }
+
+#[test]
+fn teleport_via_limited_teleport_assets_from_coretime_to_asset_hub() {
+	let amount = ASSET_HUB_WESTEND_ED * 100;
+	let native_asset: Assets = (Parent, amount).into();
+
+	test_parachain_is_trusted_teleporter!(
+		CoretimeWestend,       // Origin
+		vec![AssetHubWestend], // Destinations
+		(native_asset, amount),
+		limited_teleport_assets
+	);
+}
+
+// TODO: Fix transfer_assets weights
+// #[test]
+// fn teleport_via_transfer_assets_from_coretime_to_asset_hub() {
+// 	let amount = ASSET_HUB_WESTEND_ED * 100;
+// 	let native_asset: Assets = (Parent, amount).into();
+
+// 	test_parachain_is_trusted_teleporter!(
+// 		CoretimeWestend,    // Origin
+// 		vec![AssetHubWestend], // Destinations
+// 		(native_asset, amount),
+// 		transfer_assets
+// 	);
+// }
+
+#[test]
+fn teleport_via_limited_teleport_assets_from_asset_hub_to_coretime() {
+	let amount = CORETIME_WESTEND_ED * 100;
+	let native_asset: Assets = (Parent, amount).into();
+
+	test_parachain_is_trusted_teleporter!(
+		AssetHubWestend,       // Origin
+		vec![CoretimeWestend], // Destinations
+		(native_asset, amount),
+		limited_teleport_assets
+	);
+}
+
+// TODO: Fix transfer_assets weights
+// #[test]
+// fn teleport_via_transfer_assets_from_asset_hub_to_coretime() {
+// 	let amount = CORETIME_WESTEND_ED * 100;
+// 	let native_asset: Assets = (Parent, amount).into();
+
+// 	test_parachain_is_trusted_teleporter!(
+// 		AssetHubWestend,          // Origin
+// 		vec![CoretimeWestend], // Destinations
+// 		(native_asset, amount),
+// 		transfer_assets
+// 	);
+// }
+
+/// Limited Teleport of native asset from System Parachain to Asset Hub
+/// shouldn't work when there is not enough balance in Asset Hub's `CheckAccount`
+#[test]
+fn limited_teleport_native_assets_from_relay_to_asset_hub_checking_acc_fails() {
```

Comment:
not needed, scenario already covered by AH teleport tests

(test name is also wrong)